src/main/java/com/elastic/support/diagnostics/commands/RunKibanaQueries.java

```diff
@@ -0,0 +1,309 @@
+            }
+        }
+        totalRetries = runQueries(client, queries, context.tempDir, 0, 0);
+
+        return totalRetries;
+    }
+
+
+    /**
+    * On this function we will use the RestEntry action object to get the URL and execute the API one first time
+    * to get the total of events defined in Kibana for that API.
+    * Once we have the total of event we can calculate the number of pages/call we will need to execute
+    * then we call getNewEntryPage to create a new RestEntry for each page.
+    *
+    * @param  RestClient client
+    * @param  List<RestEntry> queries
+    * @param  double perPage - This is the number of docusment we reques to the API (set to 100 in execute func / or n in unit test)
+    * @param  RestEntry action
+    *
+    * @return         void
+    */
+    public void getAllPages(RestClient client, List<RestEntry> queries, double perPage, RestEntry action) {
+        // get the values needed to the pagination.
+        RestResult res = client.execQuery(String.format("%s?per_page=1", action.getUrl()));
+        if (! res.isValid()) {
+            throw new DiagnosticException( res.formatStatusMessage( "Could not retrieve Kibana API pagination - unable to continue."));
+        }
+        String result   = res.toString();
+        JsonNode root   = JsonYamlUtils.createJsonNodeFromString(result);
+        double total    = root.path("total").intValue();
+        if (total > 0 && perPage > 0) {
+            // Get the first actions page
+            queries.add(getNewEntryPage(perPage, 1, action));
+            // If there is more pages add the new queries
+            if (perPage < total) {
+                double numberPages = Math.ceil(total/perPage);
+                for (int i = 2; i <= numberPages; i++) {
+                    queries.add(getNewEntryPage(perPage, i, action));
+                }
+            }
+        }
+    }
+
+    /**
+    * create a new RestEntry page for the original RestEntry action 
+    *
+    * @param  int perPage
+    * @param  int page
+    * @param  RestEntry action
+    * @return RestEntry
+    */
+    private RestEntry getNewEntryPage(double perPage, int page, RestEntry action) {
+        return new RestEntry(String.format("%s_%s", action.getName(), page), "", ".json", false, String.format("%s?per_page=%s&page=%s", action.getUrl(), perPage, page), false);
+    }
+
+
+    /**
+    * This function is executed **after** runBasicQueries
+    * Extract the information on the kibana_node_stats.json with getNodeProfile function.
+    * Within the function getRemoteSystem or getLocalSystem we set ResourceCache.addSystemCommand
+    * Return will be used as workaround to Unit test. Will be replaced in v9
+    *
+    * @param  DiagnosticContext context
+    * @return SystemCommand
+    */
+    public SystemCommand execSystemCommands(DiagnosticContext context) {
+
+        ProcessProfile nodeProfile = getNodeProfile(context.tempDir, "kibana_node_stats.json", context);
```

Comment:
```suggestion
        ProcessProfile profile = getProfile(context.tempDir, "kibana_stats.json", context);
```

Reply:
I did this change in a different commit!